scenarios/generate-synthetic-data/adversarial-simulator/rewrite/simulate_and_evaluate_rewrite.ipynb

```diff
@@ -0,0 +1,176 @@
+    "from azure.identity import DefaultAzureCredential\n",
+    "from azure.ai.resources.entities import AzureOpenAIModelConfiguration\n",
+    "from openai import AsyncAzureOpenAI\n",
+    "\n",
+    "sub = \"\"\n",
+    "rg = \"\"\n",
+    "project_name = \"\"\n",
+    "oai_client = AsyncAzureOpenAI(api_key=\"\", azure_endpoint=\"\", api_version=\"2023-12-01-preview\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Initialize the simulator and get the adversarial template"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "async_oai_chat_completion_fn = oai_client.chat.completions.create\n",
+    "client = AIClient(\n",
+    "    subscription_id=sub, resource_group_name=rg, project_name=project_name, credential=DefaultAzureCredential()\n",
+    ")\n",
+    "\n",
+    "simulator = Simulator.from_fn(fn=async_oai_chat_completion_fn, ai_client=client, model=\"gpt-4\", max_tokens=300)\n",
+    "adv_template = Simulator.get_template(\"adv_rewrite\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run the simulator"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "outputs = await simulator.simulate_async(\n",
+    "    adv_template,\n",
+    "    max_conversation_turns=1,\n",
+    "    api_call_delay_sec=10,\n",
+    "    max_simulation_results=10,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Print the outputs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for line in outputs:\n",
+    "    print(json.dumps(line, indent=2))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Store outputs in file for eval"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "eval_name = \"conv-oai-rewrite-eval\"\n",
+    "file_name = f\"\\{eval_name}.jsonl\"\n",
+    "jsonl_object = outputs.to_json_lines()\n",
+    "with Path.open(file_name, \"w\") as f:\n",
+    "    f.write(jsonl_object)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run evaluate"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tracking_uri = client.tracking_uri\n",
+    "connection = client.connections.get(\"Default_AzureOpenAI\")\n",
+    "config = AzureOpenAIModelConfiguration.from_connection(connection, model_name=\"gpt-4\", deployment_name=\"gpt-4\")\n",
+    "data_path = Path.cwd() / f\"\\{eval_name}.jsonl\"\n",
+    "output_path = Path.cwd() / \"downloaded_artifacts\" / \"remote\"\n",
+    "result = evaluate(\n",
```

Comment:
the output is generated by to_jsons_lines() -- therefore it's expected to be in chat format. 
In evaluation fucntion, task_type should be "chat". 
"gpt_similarity" is not supported in chat scenario.

The supported metricf list for chat is ["hate_unfairness", "violence", "self_harm", "sexual", "gpt_coherence", "gpt_fluency", "gpt_groundedness", "gpt_relevance", "gpt_retrieval_score"], the last 3 metrics require context (documents) in assisitant role, therefore they may fail if there's no context and no "documents" under "context".